src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
I'll try to describe the use case for CurlHandler.SendAsync to set some context.

Expectation of caller of CurlHandler.SendAsync (note that this is not caller of HttpClient.SendAsync)
1. A task should be returned
2. The task should be marked complete once headers are received
3. After task has completed, the caller can optionally call response.Content.Read to retrieve the response body.

Behaviour of libcurl:
1. Whenever response headers or response body is available, a callback is invoked which has to buffer the data somewhere (or block it is consumed). The callback has knowledge of whether it was invoked for header or for body.
2. The curl_easy_perform API (there is also an async variant) does not return completion till headers AND body are received

So the TCS to used send back a task to the caller which is marked complete in StartRequest. The StartNew can be removed though.